pkg-r/R/client_state.R

```diff
@@ -74,8 +73,7 @@ method(client_set_state, S7::new_S3_class(c("Chat", "R6"))) <-
 
     replayed_turns <- lapply(
       recorded_turns,
-      ellmer::contents_replay,
-      chat = client
```

Comment:
We need to pass in tools = client$get_tools() 

(Or something similar.)

Reply:
This PR #79 gets bookmarking working (without tool calling). Those changes in #80 get bookmarking working with tool calling.

Reply:
Do you remember if that changed worked for you as-is? I'm trying to re-apply the patch and the call to `chat_enable_bookmarking()` in `chat_mod_server()` leads to

```
Error in chat_enable_bookmarking("chat", client, session = session) : 
  Error: Shiny bookmarking is not enabled. Please enable bookmarking in your
  Shiny app either by calling `shiny::enableBookmarking("server")` or by
  setting the parameter in `shiny::shinyApp(enableBookmarking = "server")`
```

And if I add `shiny::enableBookmarking("server")` to the server function of `chat_app()` or `chat_mod_server()`, I'm getting

```
Error in session$onBookmarked(fun) : 
  onBookmarked() can't be used in a module.
```

Reply:
Ohh hmm, yea. I don't think I had tested yet with the module approach. I think the way go about it properly would be to first check if bookmarking is enabled?

```r
if (!is.null(shiny::getShinyOption("bookmarkStore"))) {
   chat_enable_bookmarking("chat", client, session = session)
}
```